### PR TITLE
feat-1346: Rewrite chat service client on FE

### PR DESCRIPTION
**Pull request by SamTremko**

### CODE REVIEW

File: apps/mobile/src/components/DebugScreen/components/SimulateMissingOfferInbox.tsx [hunk 1]
```diff
@@ -32,9 +33,9 @@ function SimulateMissingOfferInbox(): JSX.Element {
       Alert.alert('No inbox for that offer found in state')
       return
     }
-    await store
-      .get(apiAtom)
-      .chat.deleteInbox({keyPair: inbox.inbox.privateKey})()
+    await effectToTaskEither(
```
Comment (kaladivo) [inline apps/mobile/src/components/DebugScreen/components/SimulateMissingOfferInbox.tsx]: Effect.runPromise()

File: apps/mobile/src/components/DebugScreen/index.tsx [hunk 2]
```diff
@@ -477,18 +477,17 @@ function DebugScreen(): JSX.Element {
               variant="primary"
               size="small"
               text="Delete user inbox"
-              onPress={() => {
-                void pipe(
-                  deleteInbox(session.privateKey),
-                  T.map((result) => {
+              onPress={() =>
+                deleteInbox(session.privateKey).pipe(
```
Comment (kaladivo) [inline apps/mobile/src/components/DebugScreen/index.tsx:481]: Effect.runPromise / runFork

File: apps/mobile/src/state/chat/atoms/sendMessageToChatsInBatchActionAtom.ts [hunk 3]
```diff
@@ -31,7 +32,7 @@ const sendMessageToChatsInBatchActionAtom = atom(
     set,
     {messageData, chats, isTerminationMessage}: Params
   ): TE.TaskEither<
-    ExtractLeftTE<ReturnType<typeof sendMessagesBatch>>,
+    ExtractErrorFromEffect<ReturnType<typeof sendMessagesBatch>>,
     boolean
   > => {
```
Comment (kaladivo) [inline apps/mobile/src/state/chat/atoms/sendMessageToChatsInBatchActionAtom.ts]: Check if this function is still in use somewhere after rebase. If not delete this...

File: packages/resources-utils/src/chat/sendMessagesBatch.ts [hunk 4]
```diff
@@ -119,39 +121,46 @@ function createInboxInBatch(
...
-      pipe(
+  return Effect.gen(function* (_) {
+    const keyPairs = inboxes.map((one) => one.inboxKeypair)
+
+    const signedChallenge = yield* _(
+      taskEitherToEffect(generateSignedChallengeBatch(api)(keyPairs))
+    )
+
+    const toSend = yield* _(
+      effectPipe(
```
Comment (kaladivo) [inline packages/resources-utils/src/chat/sendMessagesBatch.ts]: Use pipe from effect package only (even for chaining fp-ts functions)

File: packages/rest-api/src/services/chat/index.ts
Comment (kaladivo) [inline packages/rest-api/src/services/chat/index.ts:1]: Check if zod package can be removed from rest-api/package.json

File: packages/rest-api/src/services/chat/utils.ts [hunk 5]
```diff
@@ -1,84 +1,58 @@
...
-        crypto.ecdsa.ecdsaSign({
-          privateKey: privateKey.privateKeyPemBase64,
-          challenge,
-        }),
-      (e) => ({_tag: 'ErrorSigningChallenge', error: e}) as const
-    )
-}
-
-function generateChallenge({axiosInstance}: {axiosInstance: AxiosInstance}) {
+function generateChallengeE({
```
Comment (kaladivo) [inline packages/rest-api/src/services/chat/utils.ts]: removeE

File: packages/rest-api/src/utils.ts
Comment (kaladivo) [inline packages/rest-api/src/utils.ts:1]: Try to remove axios from rest-api/package.json

**Review by kaladivo — APPROVED** [review]

File: packages/rest-api/src/services/chat/index.ts
Comment (SamTremko) [reply packages/rest-api/src/services/chat/index.ts:1]: Unfortunately this cannot be done after I checked, or I would not do this now, as there are some zod interfaces leftovers in other services and removing them will cascade in changes of Session type and that brings changes in pipes for mmkv storage and so on ... And I think that's out of the scope of this task.

**Pull request merged**